kevm-pyk/src/kevm_pyk/cli.py

```diff
@@ -32,382 +49,783 @@ def node_id_like(s: str) -> NodeIdLike:
+        return {'depth': None}
+
+    @staticmethod
+    def args(parser: ArgumentParser) -> ArgumentParser:
+        parser.add_argument('--depth', default=None, type=int, help='Maximum depth to execute to.')
+        return parser
+
+
+class KompileSpecOptions(LoggingOptions, KOptions, KompileOptions):
+    main_file: Path
+    target: KompileTarget
+    output_dir: Path
+    debug_build: bool
+
+    @staticmethod
+    def default() -> dict[str, Any]:
+        return {
+            'debug_build': False,
+            'output_dir': Path(),
+            'target': KompileTarget.HASKELL,
+        }
+
+    @staticmethod
+    def parser(base: _SubParsersAction) -> _SubParsersAction:
+        base.add_parser(
+            'kompile-spec',
+            help='Kompile KEVM specification.',
+            parents=[KompileSpecOptions.all_args()],
+        )
+        return base
+
+    @staticmethod
+    def args(parser: ArgumentParser) -> ArgumentParser:
+        parser.add_argument('main_file', type=file_path, help='Path to file with main module.')
+        parser.add_argument('--target', type=KompileTarget, help='[haskell|maude]')
+        parser.add_argument(
+            '-o', '--output-definition', type=Path, dest='output_dir', help='Path to write kompiled definition to.'
+        )
+        parser.add_argument(
+            '--debug-build', dest='debug_build', default=None, help='Enable debug symbols in LLVM builds.'
+        )
+        return parser
+
+
+class ShowKCFGOptions(LoggingOptions, KOptions, SpecOptions, KEVMDisplayOptions):
+    nodes: list[NodeIdLike]
+    node_deltas: list[tuple[NodeIdLike, NodeIdLike]]
+    failure_info: bool
+    to_module: bool
+    pending: bool
+    failing: bool
+    counterexample_info: bool
+
+    @staticmethod
+    def parser(base: _SubParsersAction) -> _SubParsersAction:
+        base.add_parser(
+            'show-kcfg',
+            help='Print the CFG for a given proof.',
+            parents=[ShowKCFGOptions.all_args()],
+        )
+        return base
+
+    @staticmethod
+    def default() -> dict[str, Any]:
+        return {
+            'failure_info': False,
+            'to_module': False,
+            'pending': False,
+            'failing': False,
+            'counterexample_info': False,
+        }
+
+    @staticmethod
+    def args(parser: ArgumentParser) -> ArgumentParser:
+        parser.add_argument(
+            '--node',
+            type=node_id_like,
+            dest='nodes',
+            default=[],
+            action='append',
+            help='List of nodes to display as well.',
+        )
+        parser.add_argument(
+            '--node-delta',
+            type=arg_pair_of(node_id_like, node_id_like),
+            dest='node_deltas',
+            default=[],
+            action='append',
+            help='List of nodes to display delta for.',
+        )
+        parser.add_argument(
+            '--failure-information',
+            dest='failure_info',
+            default=None,
```

Comment:
All occurrances of `default=None` can be removed (default of `default` is `None`).

Reply:
Fixed.